iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/WorkdirInstructionImpl.java

```diff
@@ -22,23 +22,24 @@
 import java.util.ArrayList;
 import java.util.List;
 import org.sonar.iac.common.api.tree.Tree;
+import org.sonar.iac.docker.tree.api.Argument;
 import org.sonar.iac.docker.tree.api.SyntaxToken;
 import org.sonar.iac.docker.tree.api.WorkdirInstruction;
 
 public class WorkdirInstructionImpl extends InstructionImpl implements WorkdirInstruction {
 
-  private final List<SyntaxToken> workdirList;
+  private final List<Argument> arguments;
 
-  public WorkdirInstructionImpl(SyntaxToken keyword, List<SyntaxToken> workdirList) {
+  public WorkdirInstructionImpl(SyntaxToken keyword, List<Argument> arguments) {
```

Comment:
You could keep the variable name, to show what the arguments are actually representing in this context. But on the other hand, I could imagine an interface for such Instructions to provide the arguments() method. Could be useful for further processing. However, fine for now IMO.

Reply:
👍 good point, I was also thinking that later we will probably have common interface/abstract class parent as we will have many instruction with this similar grammar `keyword argument+`